File: baseline/nodejs_client/strategy.js
let BaseStrategy = require('./basestrategy').BaseStrategy;

class Strategy extends BaseStrategy {
    onTick(myPassengers, myElevators, enemyPassengers, enemyElevators) {
        myElevators.forEach(elevator => {
            myPassengers.forEach(passenger => {
                if (elevator.state != 1) {
                    elevator.goToFloor(passenger.fromFloor);
                }
                if (elevator.floor == passenger.fromFloor) {
                    passenger.setElevator(elevator);
                }
            })
            if (elevator.passengers.length > 0 && elevator.state != 1) {
                elevator.goToFloor(elevator.passengers[0].destFloor)
            }
        });
    }
}

module.exports.Strategy = Strategy;

File: baseline/c_sharp_client/strategy.cs
using System;
using client;
using Api;
using System.Collections.Generic;


namespace client
{
	public class strategy : baseStrategy
	{
		public override void onTick(List<Passenger> myPassengers, List<Elevator> myElevators, List<Passenger> enemyPassengers, List<Elevator> enemyElevators)
		{

			foreach (Elevator elevator in myElevators) 
			{
				foreach (Passenger passenger in myPassengers)
				{
					if (passenger.State < 5) 
					{
						if (elevator.State != 1)
						{
							elevator.GoToFloor(passenger.FromFloor);
						}

						if (elevator.Floor == passenger.FromFloor) 
						{
							passenger.SetElevator(elevator);
						}
					}
				}
				if (elevator.Passengers.Count > 0 && elevator.State != 1) 
				{
					elevator.GoToFloor(elevator.Passengers[0].DestFloor);
				}
			}
		}
	}
}
